frontend/src/app/components/App.js

```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+
+class App extends Component {
+  render() {
+    const { addon } = this.props;
+    if (addon.restartRequired) {
+      return <p>Restart Required</p>;
+    }
+    return this.props.children;
+  }
+}
+
+
+export default connect(
```

Comment:
Ideally this is the only connected component, and the entirety of the redux store is proliferated to children via props.

It's orthogonal to this PR so I didn't implement it here, but we would likely do it like this:

```
{React.cloneElement(this.props.children, ...this.props)}
```